Lesson4/test_distance.py

```diff
@@ -0,0 +1,41 @@
+# test_distance.py
+
+import math
+import pytest
+
+
+# function to be tested - from Lesson3
+def distance_between_points(x_1, x_2, y_1, y_2):
+    if x_1 is None or x_2 is None or y_1 is None or y_2 is None:
+        raise TypeError('None is incorrect input value')
+    if type(x_1) == int and type(x_2) == int and type(y_1) == int \
+            and type(y_2) == int:
+        distance = math.sqrt((x_2-x_1)**2 + (y_2-y_1)**2)
+        return distance
+    else:
+        raise ValueError('Incorrect input value')
```

Comment:
1. jak typ się nie zgadza to powinien być `TypeError`, jak wartość jest np spoza zakresu to wtedy `ValueError`
2. Dobrze jest utrzymywać takie flow programu, że najpierw sprawdzamy wszystko, a potem wykonujemy to co jest do zrobienia. Tutaj jest najpierw sprawdzenie jednego warunku, potem jest część właściwa funkcji, a na koniec w `else` rzucanie innego wyjątku. Lepiej byłoby odwrócić to trochę, żeby najpierw bylo sprawdzanie obu warunków, a potem kod właściwy